File: VB/ButtonEditImages/Custom Editor/CustomButtonEditPainter.vb
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports System.Drawing

Namespace ButtonEditImages

    Public Class CustomButtonEditPainter
        Inherits ButtonEditPainter

        Shared Sub New()
        End Sub

        Public Sub New()
        End Sub

        Protected Overrides Sub DrawButton(ByVal viewInfo As ButtonEditViewInfo, ByVal info As EditorButtonObjectInfoArgs)
            Dim item As RepositoryItemCustomButtonEdit = TryCast(viewInfo.Item, RepositoryItemCustomButtonEdit)
            Dim args As CustomEditorButtonObjectInfoArgs = TryCast(info, CustomEditorButtonObjectInfoArgs)
            args.SetCustomImage(item)
            MyBase.DrawButton(viewInfo, info)
        End Sub
    End Class
End Namespace

File: CS/Custom Editor/CustomButtonEditPainter.cs
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ButtonEditImages {
    public class CustomButtonEditPainter : ButtonEditPainter {
        static CustomButtonEditPainter() {
        }
        public CustomButtonEditPainter() {
        }
        protected override void DrawButton(ButtonEditViewInfo viewInfo, EditorButtonObjectInfoArgs info) {
            RepositoryItemCustomButtonEdit item = viewInfo.Item as RepositoryItemCustomButtonEdit;
            CustomEditorButtonObjectInfoArgs args = info as CustomEditorButtonObjectInfoArgs;
            args.SetCustomImage(item);
            base.DrawButton(viewInfo, info);
        }
    }
}
